helloworld/client.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Runs a simple client, designed to interact with a personality.
+package helloworld
+
+import (
+	"context"
+
+	trillian "github.com/google/trillian"
+	"github.com/google/trillian/merkle/logverifier"
+	"github.com/google/trillian/merkle/rfc6962/hasher"
+
+	p "github.com/google/trillian-examples/helloworld/personality"
+)
+
+// Personality is the interface to the the Trillian log.
+type Personality interface {
+	// Append stores an entry in the log.
+	Append(context.Context, []byte) p.Chkpt
+
+	// ProveIncl proves the inclusion of a given entry.
+	ProveIncl(context.Context, p.Chkpt, []byte) *trillian.Proof
+
+	// UpdateChkpt provides a new checkpoint and proves it is consistent
+	// with an old one.
+	UpdateChkpt(context.Context, p.Chkpt) (p.Chkpt, *trillian.Proof)
+}
+
+// A client is a verifier that maintains a checkpoint as state.
+type Client struct {
+	v      logverifier.LogVerifier
+	chkpt  p.Chkpt
+	person Personality
+}
+
+// NewClient creates a new client with an empty checkpoint and a given
+// personality to talk to.
+func NewClient(prsn Personality) Client {
+	c := Client{person: prsn}
+	v := logverifier.New(hasher.DefaultHasher)
+	c.v = v
+	var rootHash []byte
+	chkpt := p.Chkpt{LogSize: 0, RootHash: rootHash}
+	c.chkpt = chkpt
+	return c
```

Comment:
```suggestion
	v := logverifier.New(hasher.DefaultHasher)
	var rootHash []byte
	chkpt := p.Chkpt{LogSize: 0, RootHash: rootHash}
	return Client{
	      person: prsn,
	      v: v,
	      chkpt: chkpt,
	}
```

It's more standard / safe to set all the fields in one construction rather than building it up.